sessiontxn/isolation/readcommitted.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package isolation
+
+import (
+	"context"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/parser/ast"
+	"github.com/pingcap/tidb/parser/terror"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/sessiontxn"
+	"github.com/pingcap/tidb/util/logutil"
+	tikverr "github.com/tikv/client-go/v2/error"
+	"github.com/tikv/client-go/v2/oracle"
+	"go.uber.org/zap"
+)
+
+type stmtState struct {
+	stmtTS            uint64
+	stmtTSFuture      oracle.Future
+	stmtUseStartTS    bool
+	onNextRetryOrStmt func() error
+}
+
+func (s *stmtState) resetStmt(useStartTS bool) error {
```

Comment:
`Reset` usually means the object is restored to an initial state and it could be resued by others. Maybe `prepareStmt` or `onNextStmt` is more appropriate here?

Reply:
done